alica_engine/src/engine/modelmanagement/factories/RoleFactory.cpp

```diff
@@ -23,11 +22,10 @@ Role* RoleFactory::create(const YAML::Node& roleNode, RoleSet* roleSet)
 
 void RoleFactory::attachReferences()
 {
-    // roleTaskReferences
     for (std::tuple<int64_t, int64_t, double> triple : Factory::roleTaskReferences) {
```

Comment:
I think we can use structured binding now.

Reply:
I wish I had are more meaningful types, not plain int64_t

Reply:
and unfortunately we should call `Factory::getElement` for each tuple element

Reply:
It's a blocking comment? It's out of scope of initial PR